src/validationinterface.cpp

```diff
@@ -75,8 +75,10 @@ CMainSignals& GetMainSignals()
     return g_signals;
 }
 
-void RegisterValidationInterface(CValidationInterface* pwalletIn) {
-    ValidationInterfaceConnections& conns = g_signals.m_internals->m_connMainSignals[pwalletIn];
+void RegisterSharedValidationInterface(std::shared_ptr<CValidationInterface> pwalletIn) {
```

Comment:
nit: Mind identical parameter naming in the `RegisterSharedValidationInterface` declaration and definition:
`pwalletIn` vs `callbacks` ?

Reply:
Let's do that in one of Russel suggestions and avoid unnecessary changes here.